### PR TITLE
Unified test documentation: embedded controller test suite

### PR DESCRIPTION
**Pull request by sulewskiprzemyslaw**
Signed-off-by: Przemysław Sulewski <przemyslaw.sulewski@3mdeb.com>

### CODE REVIEW

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 1]
```diff
@@ -0,0 +1,1266 @@
...
+1. `BIOS_SETUP_KEY` = `F2`
+
+**Test setup**
+
+1. Proceed with the
+   [Generic test setup: firmware](../generic-test-setup/#firmware)
+
+**Test steps**
+
+1. Power on the DUT and press the `BIOS_SETUP_KEY` to enter the setup menu
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:228]:
```suggestion
1. Power on the DUT and press the `BIOS_SETUP_KEY` to enter the setup menu.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 2]
```diff
@@ -0,0 +1,1266 @@
...
+
+**Test steps**
+
+1. Press the touchpad on/off key and try to use the touchpad.
+1. Press the touchpad on/off key once again and try to use the touchpad again.
+
+**Expected result**
+
+1. Pressing the hotkey once should deactivate the touchpad (touchpad should be
+   completely inoperable)
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:455]:
```suggestion
   completely inoperable).
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 3]
```diff
@@ -0,0 +1,1266 @@
...
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Press the touchpad on/off key and try to use the touchpad.
+1. Press the touchpad on/off key once again and try to use the touchpad again.
+
+**Expected result**
+
+1. Pressing the hotkey once should deactivate the touchpad (touchpad should be
+   completely inoperable)
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:484]:
```suggestion
   completely inoperable).
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 4]
```diff
@@ -0,0 +1,1266 @@
...
+1. Pressing the hotkey again should reactivate the touchpad
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:485]:
```suggestion
1. Pressing the hotkey again should reactivate the touchpad.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 5]
```diff
@@ -0,0 +1,1266 @@
...
+**Test setup**
+
+1. Proceed with the [Common](#common) section.
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Press the display on/off hotkey once and note the effect
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:507]:
```suggestion
1. Press the display on/off hotkey once and note the effect.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 6]
```diff
@@ -0,0 +1,1266 @@
...
+1. Press any key on the keyboard and note the effect
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:508]:
```suggestion
1. Press any key on the keyboard and note the effect.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 7]
```diff
@@ -0,0 +1,1266 @@
...
+
+**Expected result**
+
+1. Pressing the hotkey once should turn the internal LCD panel off
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:512]:
```suggestion
1. Pressing the hotkey once should turn the internal LCD panel off.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 8]
```diff
@@ -0,0 +1,1266 @@
...
+1. Pressing any key on the keyboard should power the internal LCD panel back on
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:513]:
```suggestion
1. Pressing any key on the keyboard should power the internal LCD panel back on.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 9]
```diff
@@ -0,0 +1,1266 @@
...
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Press the mute hotkey once and check the volume indicator in the bottom right
+   part of the screen.
+1. Press the mute hotkey once and check the volume indicator again.
+
+**Expected result**
+
+1. Pressing the hotkey once should mute the device
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:597]:
```suggestion
1. Pressing the hotkey once should mute the device.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 10]
```diff
@@ -0,0 +1,1266 @@
...
+1. Pressing the hotkey again should reenable the sound
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:598]:
```suggestion
1. Pressing the hotkey again should reenable the sound.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 11]
```diff
@@ -0,0 +1,1266 @@
...
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Press the volume down hotkey once and note the effects.
+
+**Expected result**
+
+1. Pressing the hotkey should decrease the volume of the currently enabled audio
+   output.
+1. Each keypress should cause a volume down notification to appear in the middle
+   of the screen
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:685]:
```suggestion
   of the screen.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 12]
```diff
@@ -0,0 +1,1266 @@
...
+
+1. Proceed with the [Common](#common) section.
+1. Install `libinput-tools` on the DUT.
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Run `sudo libinput debug-events --show-keycodes` in the terminal
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:795]:
```suggestion
1. Run `sudo libinput debug-events --show-keycodes` in the terminal.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 13]
```diff
@@ -0,0 +1,1266 @@
...
+1. Press the display switch hotkey once and note the effect
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:796]:
```suggestion
1. Press the display switch hotkey once and note the effect.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 14]
```diff
@@ -0,0 +1,1266 @@
...
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Open the terminal and run the command: `watch -n1 lsusb`.
+1. Press the camera on/off hotkey twice and note the effect after each keypress.
+
+**Expected result**
+
+1. Pressing the hotkey once should make the `Chicony Electronics Co., Ltd Chicony USB2.0 Camera`
+   device disappear from the output of `lsusb`
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:974]:
```suggestion
   device disappear from the output of `lsusb`.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 15]
```diff
@@ -0,0 +1,1266 @@
...
+1. Pressing the hotkey again should make the USB device reappear
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:975]:
```suggestion
1. Pressing the hotkey again should make the USB device reappear.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 16]
```diff
@@ -0,0 +1,1266 @@
...
+**Test setup**
+
+1. Proceed with the [Common](#common) section.
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Locate and open the power menu in the upper right corner of the screen
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:1027]:
```suggestion
1. Locate and open the power menu in the upper right corner of the screen.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 17]
```diff
@@ -0,0 +1,1266 @@
...
+1. Enter Settings and navigate to the Wi-Fi panel
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:1028]:
```suggestion
1. Enter Settings and navigate to the Wi-Fi panel.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 18]
```diff
@@ -0,0 +1,1266 @@
...
+1. Note the current state of the Airplane Mode switch
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:1029]:
```suggestion
1. Note the current state of the Airplane Mode switch.
```

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 19]
```diff
@@ -0,0 +1,1266 @@
...
+1. Press the flight mode hotkey twice and note the result of each keypress
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:1030]:
```suggestion
1. Press the flight mode hotkey twice and note the result of each keypress.
```

**Review by damiankaras — CHANGES_REQUESTED** [review]

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md (hunk 1 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:228]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 20]
```diff
@@ -0,0 +1,1266 @@
...
+
+**Test setup**
+
+1. Proceed with the
+   [Generic test setup: firmware](../generic-test-setup/#firmware)
+
+**Test steps**
+
+1. Power on the DUT and press the `BIOS_SETUP_KEY` to enter the setup menu
+1. Use the arrow keys and the Enter key to navigate the menus
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:229]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md (hunk 15 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:975]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md (hunk 19 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:1030]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 21]
```diff
@@ -0,0 +1,1266 @@
...
+
+1. Proceed with the [Common](#common) section.
+1. Install `libinput-tools` on the DUT.
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Run `sudo libinput debug-events --show-keycodes` in the terminal
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:323]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 22]
```diff
@@ -0,0 +1,1266 @@
...
+1. Verify that pressing the play/pause key generates a `KEY_PLAYPAUSE` event
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:324]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 23]
```diff
@@ -0,0 +1,1266 @@
...
+**Test setup**
+
+1. Proceed with the [Common](#common) section.
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Press the cooling mode hotkey (Fn + 1) once and note the effect
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:378]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 24]
```diff
@@ -0,0 +1,1266 @@
...
+1. Press the cooling mode hotkey once again and note the effect
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:379]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 25]
```diff
@@ -0,0 +1,1266 @@
...
+
+**Expected result**
+
+1. Pressing the hotkey once should activate the cooling mode (fans should
+   spin up to their maximum speed)
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:384]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 26]
```diff
@@ -0,0 +1,1266 @@
...
+
+1. Proceed with the [Common](#common) section.
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Press the cooling mode hotkey (Fn + 1) once and note the effect
+1. Press the cooling mode hotkey once again and note the effect
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:409]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 27]
```diff
@@ -0,0 +1,1266 @@
...
+
+**Expected result**
+
+1. Pressing the hotkey once should activate the cooling mode (fans should
+   spin up to their maximum speed)
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:414]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 28]
```diff
@@ -0,0 +1,1266 @@
...
+1. Pressing the hotkey again should deactivate the cooling mode (fans should
+   return to normal)
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:416]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md [hunk 29]
```diff
@@ -0,0 +1,1266 @@
...
+**Test steps**
+
+1. Press the touchpad on/off key and try to use the touchpad.
+1. Press the touchpad on/off key once again and try to use the touchpad again.
+
+**Expected result**
+
+1. Pressing the hotkey once should deactivate the touchpad (touchpad should be
+   completely inoperable)
+1. Pressing the hotkey again should reactivate the touchpad
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:456]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md (hunk 5 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:507]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md (hunk 6 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio.md:508]: Improved

**Pull request merged**